### PR TITLE
Return cost components as json; add getter cdb2_get_info()

### PR DESCRIPTION
**Pull request by adizaimi**
Include statement cost and extended cost as part of lastresponse
so we avoid extra call to the db to get cost.

* Always return numeric value of cost
* If get_cost is set, return extended cost info--cost components
* `cdb2sql --cost` prints extended cost as json string using cdb2api
  getter function cdb2_get_info()

Example json extended cost from cdb2sql:

```
> cdb2sql adidb --cost 'select moredata from t2 where uid in (select uid from t1 where i=2)' | tail +2
{"Cost":32.40,"NumRows":1,
 "Components": [
   {"Type":"Table Scan","Table":"t1","NumFind":1,"NumNextOrPrev":1},
   {"Type":"Table Scan","Table":"t2","NumFind":1},
   {"Type":"Index Lookup","Index":0,"Table":"t2","NumFind":1,"NumNextOrPrev":1},
   {"Type":"Temp Index Lookup","NumFind":1,"NumPrevOrNext":1,"NumWrites":1 }
]}
```

Signed-off-by: Adi Zaimi <azaimi@bloomberg.net>

### CODE REVIEW

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ cdb2api/cdb2api.c @@
+ Please port the change to cdb2jdbc if necessary.
+ If you add any internal configuration state to this file, please update the reset_the_configuration() function as well to include it.
```

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/formatting-a-pr.html'>Use clang-format</a> or <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/77d50961/format.diff'>view diff</a> &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>1/387 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>sql_cost</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/77d50961/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/77d50961/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2601'>Resubmit</a></p>

File: tools/cdb2sql/cdb2sql.cpp [hunk 1]
```diff
@@ -1589,6 +1589,13 @@ static int run_statement(const char *sql, int ntypes, int *types,
             printf("Effects not sent by comdb2 server. \n");
         }
     }
+
+    if (docost) {
+        const char *str = cdb2_get_cost(cdb2h);
+        if(str) {
+            printf("%s\n", str);
```
Comment (riverszhang89) [inline tools/cdb2sql/cdb2sql.cpp:1596]: puts

File: tools/cdb2sql/cdb2sql.cpp [hunk 2]
```diff
@@ -1589,6 +1589,13 @@ static int run_statement(const char *sql, int ntypes, int *types,
             printf("Effects not sent by comdb2 server. \n");
         }
     }
+
+    if (docost) {
```
Comment (riverszhang89) [inline tools/cdb2sql/cdb2sql.cpp:1593]: We need to use the same checks here. We probably do not want to print cost if there's a bad rcode. We do not want to print cost twice if a custom `report_costs` is defined.

**Review by riverszhang89 — CHANGES_REQUESTED** [review]

File: tools/cdb2sql/cdb2sql.cpp (hunk 2 above)
Comment (adizaimi) [reply tools/cdb2sql/cdb2sql.cpp:1593]: We don't get here in rc is not OK_DONE (line 1574).

File: tools/cdb2sql/cdb2sql.cpp (hunk 1 above)
Comment (adizaimi) [reply tools/cdb2sql/cdb2sql.cpp:1596]: will do

**Review by nirbhayc — DISMISSED** [review]
Looks good.

File: cdb2api/cdb2api.h [hunk 3]
```diff
@@ -201,6 +201,7 @@ int cdb2_clone(cdb2_hndl_tp **hndl, cdb2_hndl_tp *c_hndl);
 int cdb2_next_record(cdb2_hndl_tp *hndl);
 
 int cdb2_get_effects(cdb2_hndl_tp *hndl, cdb2_effects_tp *effects);
+const char *cdb2_get_cost(cdb2_hndl_tp *hndl);
```
Comment (riverszhang89) [inline cdb2api/cdb2api.h:204]: Maybe just declare it in cdb2sql.cpp. I don't think this function will get a lot of usage except from cdb2sql.

**Comment by adizaimi** [comment]
Closing to reopen a new PR with same branch.

**Pull request closed without merging**